Rework OIDC page intros. (#2925)
buildkite/docs: pages/platform/_oidc_introduction.md

> [Open ID Connect (OIDC)](https://openid.net/developers/how-connect-works/) is an authentication protocol based on the [OAuth 2.0 framework](https://auth0.com/docs/authenticate/protocols/oauth/) that allows an _OIDC token_, containing metadata (or _claims_) about a user or object, which is issued by one system or service (as part of one or more actions), to be consumed and authenticated by another, such as a service delivered by a third-party or another offered by the same organization. An _OIDC policy_ configured on this other service defines which OIDC tokens (based on their claims) are permitted to perform the actions. If the OIDC token's claims match those of the OIDC policy configured in the other service, the token is authenticated and the service issuing the token is permitted to perform its actions on the other service.

Reviewer: This is accurate, though the opening sentence is a bit complex and hard to parse. Not that OIDC is ever easy to understand.

I do wonder if the word "signed" should stay in there somewhere. Anyone can make a _claim_, that doesn't really mean anything. A _signed_ claim is something you can do something with.

Here's my attempt to unravel the first sentence a bit. I don't know if it's better, but maybe it's useful food for thought.

```suggestion
[Open ID Connect (OIDC)](https://openid.net/developers/how-connect-works/) is an authentication protocol based on the [OAuth 2.0 framework](https://auth0.com/docs/authenticate/protocols/oauth/). With OIDC, one system or service issues an _OIDC token_ containing signed metadata (or _claims_) about a user or object. This token can be consumed by another service (which may be offered by a third-party or by the same organization) to authenticate the user or object. An _OIDC policy_ configured on this other service defines which OIDC tokens (based on their claims) are permitted to perform the actions. If the OIDC token's claims match those of the OIDC policy configured in the other service, the token is authenticated and the service issuing the token is permitted to perform its actions on the other service.
```

Author: Love this! Thanks @tessereth ! This reads a lot better 😄
